File: core/src/main/java/pcb/uwu/units/derived/scalar/SolidAngleUnit.java
package pcb.uwu.units.derived.scalar;

import pcb.uwu.core.BaseUnit;
import pcb.uwu.core.UnitCounter;

import java.util.function.Function;

public class SolidAngleUnit extends BaseUnit {

	protected SolidAngleUnit() {
		super(0, "", "", "", Function.identity(), Function.identity());
	}

	@Override
	public Class<? extends BaseUnit> getBaseUnitType() {
		return null;
	}

	@Override
	public UnitCounter getUnitCounter() {
		return new UnitCounter();
	}
}

File: dimension/src/main/java/pcb/uwu/unit/derived/geometry/SolidAngleUnit.kt
package pcb.uwu.unit.derived.geometry

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.AngleUnit
import pcb.uwu.unit.base.LuminousIntensityUnit
import pcb.uwu.unit.derived.optics.LuminousFluxUnit

open class SolidAngleUnit
    : CompositeUnit
{
    @Suppress("ConvertSecondaryConstructorToPrimary")
    constructor(angleUnit1: AngleUnit,
                angleUnit2: AngleUnit)
            : super(UnitCounter()
                        .major(angleUnit1)
                        .major(angleUnit2))

    constructor(luminousFluxUnit: LuminousFluxUnit,
                luminousIntensityUnit: LuminousIntensityUnit)
            : super(UnitCounter()
                        .major(luminousFluxUnit)
                        .minor(luminousIntensityUnit))

}
